pkg/autoupdate/tuf/read_only_library.go

```diff
@@ -0,0 +1,269 @@
+package tuf
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/autoupdate"
+)
+
+// readOnlyLibrary provides a read-only view into the updates libraries for all
+// binaries.
+type readOnlyLibrary struct {
```

Comment:
I don't wholly understand the motivation for splitting this out. It feels like it's adding a lot of complexity, since the RO and RW versions are inherently intertwinned.

Reply:
I went to make a diagram to explain why I've split this out, but in the process I've realized I've split out the wrong bit.

The original motivation was to remove the osquerier dependency, because `findNewest` needs to run before we have an osquery client. However, we only need the osquerier to get the current running version, which we only need to run the routine to tidy the library (so that we don't delete the currently-running version).

I'll think on it a little more, but I think the direction I'm leaning is to move the osquerier dependency out of the library altogether and re-combine.

Reply:
Happy to rubber duck or brainstorm if it would help.

Reply:
I've just pushed this update, if you'd like to take a quick look and have thoughts! Wondering if I should maybe move some of the install version stuff into a new file, or maybe it's fine to keep in the library.